context/QuestsProvider.tsx

```diff
@@ -94,8 +94,8 @@ export const QuestsContextProvider = ({
 
   useMemo(() => {
     getTrendingQuests(hexToDecimal(address)).then(
-      (data: QuestDocument[] | QueryError) => {
-        if ((data as QueryError).error) return;
+      (data: QuestDocument[] | QueryError | undefined) => {
```

Comment:
why a undefined check here?

Reply:
well if you see in our getTrendingQuests function we have a catch block and we are just doing a console log and not returning any value. so typescript assumes that the function has a return type of void as well as return type of QuestDocument. Because of that i was having an error in typescript at that line saying data might be undefined.

Reply:
Can you return an error here instead like query error or something to handle this? I personally would like to avoid the undefined values to come till the state variable

Reply:
@ayushtom made changes